dotCMS/src/main/java/com/dotcms/contenttype/model/type/VanityUrlContentType.java

```diff
@@ -31,73 +31,73 @@
-        Field forwardToField = ImmutableCustomField.builder().name(FORWARD_TO_FIELD_NAME)
-                .variable(FORWARD_TO_FIELD_VAR)
-                .dataType(DataTypes.TEXT).fixed(Boolean.TRUE).indexed(Boolean.TRUE)
-                .values("$velutil.mergeTemplate('/static/content/file_browser_field_render.vtl')")
-                .required(Boolean.TRUE).sortOrder(order++).listed(Boolean.TRUE).build();
-        Field orderField = ImmutableTextField.builder().name(ORDER_FIELD_NAME)
-                .variable(ORDER_FIELD_VAR)
-                .dataType(DataTypes.INTEGER).required(Boolean.TRUE).fixed(Boolean.TRUE)
-                .indexed(Boolean.TRUE)
-                .sortOrder(order++).searchable(Boolean.TRUE).defaultValue("0").build();
-        return ImmutableList
-                .of(titleField, siteField, uriField, forwardToField, actionField, orderField);
-    }
+	@Override
+	public List<Field> requiredFields() {
+		int order = 0;
+		Field titleField = ImmutableTextField.builder().name(TITLE_FIELD_NAME)
+				.variable(TITLE_FIELD_VAR)
+				.dataType(DataTypes.TEXT).required(Boolean.TRUE).listed(Boolean.TRUE)
+				.indexed(Boolean.TRUE)
+				.sortOrder(order++).fixed(Boolean.TRUE).searchable(Boolean.TRUE).build();
+		Field siteField = ImmutableCustomField.builder().name(SITE_FIELD_NAME)
+				.variable(SITE_FIELD_VAR)
+				.dataType(DataTypes.TEXT).fixed(Boolean.TRUE).indexed(Boolean.TRUE)
+				.values("$velutil.mergeTemplate('/static/content/site_selector_field_render.vtl')")
+				.required(Boolean.TRUE).sortOrder(order++).listed(Boolean.FALSE).build();
+		Field uriField = ImmutableTextField.builder().name(URI_FIELD_NAME).variable(URI_FIELD_VAR)
+				.dataType(DataTypes.TEXT).indexed(Boolean.TRUE).searchable(Boolean.TRUE)
+				.required(Boolean.TRUE)
+				.sortOrder(order++).fixed(Boolean.TRUE).listed(Boolean.TRUE).build();
+		Field actionField = ImmutableSelectField.builder().name(ACTION_FIELD_NAME)
+				.variable(ACTION_FIELD_VAR)
+				.required(Boolean.TRUE).fixed(Boolean.TRUE).indexed(Boolean.TRUE)
+				.searchable(Boolean.TRUE).sortOrder(order++)
+				.dataType(DataTypes.INTEGER)
+				.values("200 - Forward|200\r\n301 - Permanent Redirect|301\r\n302 - Temporary Redirect|302\r\n401 - Auth Required|401\r\n403 - Auth Failed|403\r\n404 - Missing|404\r\n500 - Error|500")
+				.build();
+		Field forwardToField = ImmutableCustomField.builder().name(FORWARD_TO_FIELD_NAME)
+				.variable(FORWARD_TO_FIELD_VAR)
+				.dataType(DataTypes.TEXT).fixed(Boolean.TRUE).indexed(Boolean.TRUE)
+				.values("$velutil.mergeTemplate('/static/content/file_browser_field_render.vtl')")
+				.required(Boolean.TRUE).sortOrder(order++).listed(Boolean.TRUE).build();
+		Field orderField = ImmutableTextField.builder().name(ORDER_FIELD_NAME)
+				.variable(ORDER_FIELD_VAR)
+				.dataType(DataTypes.INTEGER).required(Boolean.TRUE).fixed(Boolean.TRUE)
+				.indexed(Boolean.TRUE)
+				.sortOrder(order++).searchable(Boolean.TRUE).defaultValue("0").build();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "order". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)